src/storage/store/column.cpp

```diff
@@ -582,10 +648,52 @@ void Column::prepareCommitForChunk(Transaction* transaction, node_group_idx_t no
     }
 }
 
+void Column::prepareCommitForChunk(Transaction* transaction, node_group_idx_t nodeGroupIdx,
+    offset_t csrOffset, ColumnChunk* chunk, offset_t dataOffset, length_t numValues) {
+    auto currentNumNodeGroups = metadataDA->getNumElements(transaction->getType());
+    auto isNewNodeGroup = nodeGroupIdx >= currentNumNodeGroups;
+    if (isNewNodeGroup) {
+        commitColumnChunkOutOfPlace(
+            transaction, nodeGroupIdx, isNewNodeGroup, csrOffset, chunk, dataOffset, numValues);
+    } else {
+        bool didInPlaceCommit = false;
+        // If this is not a new node group, we should first check if we can perform in-place commit.
+        if (canCommitInPlace(transaction, nodeGroupIdx, csrOffset, chunk, dataOffset, numValues)) {
+            commitColumnChunkInPlace(nodeGroupIdx, csrOffset, chunk, dataOffset, numValues);
+            didInPlaceCommit = true;
+        } else {
+            commitColumnChunkOutOfPlace(
+                transaction, nodeGroupIdx, isNewNodeGroup, csrOffset, chunk, dataOffset, numValues);
+        }
+        if (nullColumn) {
+            if (nullColumn->canCommitInPlace(transaction, nodeGroupIdx, csrOffset,
+                    chunk->getNullChunk(), dataOffset, numValues)) {
+                nullColumn->commitColumnChunkInPlace(
+                    nodeGroupIdx, csrOffset, chunk->getNullChunk(), dataOffset, numValues);
+            } else if (didInPlaceCommit) {
+                nullColumn->commitColumnChunkOutOfPlace(transaction, nodeGroupIdx, isNewNodeGroup,
+                    csrOffset, chunk->getNullChunk(), dataOffset, numValues);
+            }
+        }
+    }
+}
+
 bool Column::canCommitInPlace(Transaction* transaction, node_group_idx_t nodeGroupIdx,
     LocalVectorCollection* localChunk, const offset_to_row_idx_t& insertInfo,
     const offset_to_row_idx_t& updateInfo) {
     auto metadata = getMetadata(nodeGroupIdx, transaction->getType());
+    auto maxOffset = 0u;
+    for (auto& [offset, rowIdx] : insertInfo) {
+        if (offset > maxOffset) {
+            maxOffset = offset;
+        }
+    }
+    if ((metadata.compMeta.numValues(BufferPoolConstants::PAGE_4KB_SIZE, *dataType) *
+            metadata.numPages) <= maxOffset) {
+        // Note that for constant compression, `metadata.numPages` will be equal to 0. Thus, this
+        // function will always return false.
+        return false;
```

Comment:
Don't we want constant compression to be able to do in-place commits if the values happen to be identical to the existing ones? Or is that handled somewhere else?

Reply:
Nice catch. Changed the logic here to enable potential in-place commits for constant compression.